File: Code Boiler/Dart/signup.dart
import 'dart:convert';
import 'dart:io';
import 'dart:math';

class SignUp {
  String? firstName;
  String? lastName;
  String? email;
  String? password;
  String? accountNumber;

  SignUp() {
    print("Enter your first name: ");
    firstName = stdin.readLineSync();
    print("Enter your last name: ");
    lastName = stdin.readLineSync();
    print("Enter your email: ");
    email = stdin.readLineSync();
    print("Enter your password: ");
    password = stdin.readLineSync();
    accountNumber = firstName![0] +
        lastName![0] +
        Random().nextInt(1 << 32).toRadixString(16);
  }

  bool validateInput() {
    if (!email!.endsWith("@example.com")) {
      print("Invalid email address");
      return false;
    }
    if (password!.length < 8) {
      print("Password too short");
      return false;
    }
    return true;
  }

  void register() {
    if (!validateInput()) {
      return;
    }
    print("Registration successful! Your account number is: $accountNumber");
    exit(0);
  }
}

void main() {
  SignUp user = SignUp();
  user.register();
}

File: Code Boiler/python/signup.py
import secrets 
import sys 


# PURPOSE OF THIS IS FOR USER TO SIGN UP FOR AN ACCOUNT 


class SignUp:
    def __init__(self):
        self.first_name = input("Enter your first name: ")
        self.last_name = input("Enter your last name: ")
        self.email = input("Enter your email: ")
        self.password = input("Enter your password: ")
        self.account_number = self.first_name[0] + self.last_name[0] + secrets.token_hex(4)

    def validate_input(self):
        if not self.email.endswith('@example.com'):
            print("Invalid email address")
            return False
        if len(self.password) < 8:
            print("Password too short")
            return False
        return True

    def register(self):
        if not self.validate_input():
            return
        print("Registration successful! Your account number is: ", self.account_number)
        sys.exit() 

user = SignUp()
user.register()